main/lsp/LSPPreprocessor.h

```diff
@@ -1,20 +1,29 @@
 #ifndef RUBY_TYPER_LSP_LSPPREPROCESSOR_H
 #define RUBY_TYPER_LSP_LSPPREPROCESSOR_H
 
-#include "core/core.h"
 #include "main/lsp/LSPConfiguration.h"
 #include "main/lsp/LSPMessage.h"
 #include <deque>
 
 namespace sorbet::realmain::lsp {
 
-/** Used to store the state of LSP's internal request queue.  */
-struct QueueState {
+class LSPTask;
+
+struct MessageQueueState {
```

Comment:
The `MessageQueueState` is still used to enqueue raw messages from the client and watchman for preprocessing.

The `TaskQueueState` is what the `LSPPreprocessor` operates on.